aries_vcx_core/src/wallet/base_wallet.rs

```diff
@@ -26,18 +28,28 @@ pub trait BaseWallet: std::fmt::Debug + Send + Sync {
 
     // ---- records
 
-    async fn add_wallet_record(&self, xtype: &str, id: &str, value: &str, tags_json: Option<&str>)
-        -> VcxCoreResult<()>;
+    async fn add_wallet_record(
+        &self,
+        xtype: &str,
+        id: &str,
+        value: &str,
+        tags: Option<HashMap<String, String>>,
+    ) -> VcxCoreResult<()>;
 
-    async fn get_wallet_record(&self, xtype: &str, id: &str, options_json: &str) -> VcxCoreResult<String>;
+    async fn get_wallet_record(&self, xtype: &str, id: &str, options: &str) -> VcxCoreResult<String>;
```

Comment:
The `options` could actually be well typed, but its type depends on what kind of options the underlying implementation accepts.

This is slightly more advanced, but we could use associated types here. You would defined something like

```
 pub trait BaseWallet: std::fmt::Debug + Send + Sync {
    type GetRecordOptions;
    type AddRecordOptions;

   ...
   ...
}
```
Then in particular implementation of the trait, for example in `impl IndySdkWallet `, you would inject the actual types. And of course, you would have to create those types (you can find out what the option objects should look like if you dig into `libvdrtools` code)

 So not a hard requirement, but if you could do this it would be really great!

Reply:
alright!